src/registry/number/ValueDisplay.tsx

```diff
@@ -0,0 +1,20 @@
+import type {NumberComponentSchema} from '@open-formulieren/types';
+import {useIntl} from 'react-intl';
+
+export interface ValueDisplayProps {
+  componentDefinition: NumberComponentSchema;
+  value: number | null;
+}
+
+const ValueDisplay: React.FC<ValueDisplayProps> = ({
+  // @ts-ignore
+  componentDefinition,
+  value,
+}) => {
+  const {locale} = useIntl();
+
+  if (value === null) return '';
+  return new Intl.NumberFormat(locale).format(value);
```

Comment:
can't we use https://formatjs.github.io/docs/react-intl/components/#formattednumber? (it uses the `Intl` APIs under the hood)